core/src/io.rs

```diff
@@ -229,75 +268,68 @@ impl IoHandler {
 		self.request_handler.add_subscriptions(delegate.subscriptions);
 	}
 
-	/// Handle given request synchronously - will block until response is available.
-	/// If you have any asynchronous methods in your RPC it is much wiser to use
-	/// `handle_request` instead and deal with asynchronous requests in a non-blocking fashion.
-	pub fn handle_request_sync(&self, request_str: &str) -> Option<String> {
-		let (tx, rx) = mpsc::channel();
-		self.handle_request(request_str, move |response: Option<String>| {
-			tx.send(response).expect("Receiver is never dropped.");
-		});
-		// Wait for response.
-		rx.recv().expect("Transmitting end is never dropped.")
-	}
-
-	/// Handle given request asynchronously.
-	pub fn handle_request<H: ResponseHandler<Option<String>, Option<String>> + 'static>(&self, request_str: &str, response_handler: H) {
-		self.handle(request_str, response_handler, None);
+	/// Returns a request handler to issue calls directly.
+	pub fn request_handler(&self) -> &RequestHandler {
+		&self.request_handler
 	}
 
-	fn handle<H: ResponseHandler<Option<String>, Option<String>> + 'static>(&self, request_str: &str, response_handler: H, session: Option<Session>) {
-		trace!(target: "rpc", "Request: {} in session.", request_str);
-
-		let handler = Handler::new(
+	/// Converts the handler to serializing one.
+	pub fn convert_handler<H>(response_handler: H) -> Handler<Option<String>, Option<Response>, Notification> where
+		H: ResponseHandler<Option<String>, Option<String>> + 'static
```

Comment:
Did it for `Handler<Option<String> ....>`, but can't do it for `ResponseHandler<Option<String>, Option<String>>`:
```
note: type aliases cannot be used for traits
```